packages/nodes-from-anchor/src/v01/typeNodes/TypeNode.ts

```diff
@@ -0,0 +1,106 @@
+import { KINOBI_ERROR__UNRECOGNIZED_ANCHOR_IDL_TYPE, KinobiError } from '@kinobi-so/errors';
+import {
+    booleanTypeNode,
+    bytesTypeNode,
+    definedTypeLinkNode,
+    numberTypeNode,
+    publicKeyTypeNode,
+    sizePrefixTypeNode,
+    stringTypeNode,
+    TypeNode,
+} from '@kinobi-so/nodes';
+
+import {
+    IdlV01Type,
+    IdlV01TypeCOption,
+    IdlV01TypeDefTy,
+    IdlV01TypeDefTyEnum,
+    IdlV01TypeDefTyStruct,
+    IdlV01TypeOption,
+} from '../idl';
+import { arrayTypeNodeFromAnchorV01 } from './ArrayTypeNode';
+import { enumTypeNodeFromAnchorV01 } from './EnumTypeNode';
+import { optionTypeNodeFromAnchorV01 } from './OptionTypeNode';
+import { structTypeNodeFromAnchorV01 } from './StructTypeNode';
+
+const IDL_V01_TYPE_LEAVES = [
+    'string',
+    'pubkey',
+    'bytes',
+    'bool',
+    'u8',
+    'u16',
+    'u32',
+    'u64',
+    'u128',
+    'i8',
+    'i16',
+    'i32',
+    'i64',
+    'i128',
+    'f32',
+    'f64',
+    'shortU16',
+] as const;
+
+export const typeNodeFromAnchorV01 = (idlType: IdlV01Type | IdlV01TypeDefTy): TypeNode => {
+    // Leaf.
+    if (typeof idlType === 'string' && IDL_V01_TYPE_LEAVES.includes(idlType)) {
+        if (idlType === 'bool') return booleanTypeNode();
+        if (idlType === 'pubkey') return publicKeyTypeNode();
+        if (idlType === 'string') return sizePrefixTypeNode(stringTypeNode('utf8'), numberTypeNode('u32'));
+        if (idlType === 'bytes') return sizePrefixTypeNode(bytesTypeNode(), numberTypeNode('u32'));
+        return numberTypeNode(idlType);
+    }
+
+    // Ensure eveything else is an object.
+    if (typeof idlType !== 'object') {
+        throw new KinobiError(KINOBI_ERROR__UNRECOGNIZED_ANCHOR_IDL_TYPE, {
+            idlType: JSON.stringify(idlType),
+        });
```

Comment:
nit: maybe we should add a `idlVersion` field on the context of this error. See `packages/errors/src/context.ts`.

Reply:
another nit: Since we're gonna have multiple Anchor errors now, we should create an "Anchor" group of errors starting with `KINOBI_ERROR__ANCHOR__`.

So `KINOBI_ERROR__UNRECOGNIZED_ANCHOR_IDL_TYPE` could be renamed `KINOBI_ERROR__ANCHOR__UNRECOGNIZED_IDL_TYPE`.